src/http.cc

```diff
@@ -404,7 +404,7 @@ HttpStateData::reusableReply(HttpStateData::ReuseDecision &decision)
             // HTTPbis WG verdict on this is that it is omitted from the spec due to being 'unexpected' by
             // some. The caching+revalidate is not exactly unsafe though with Squids interpretation of no-cache
             // (without parameters) as equivalent to must-revalidate in the reply.
-        } else if (rep->cache_control->hasNoCacheWithoutParameters()) {
+        } else if (rep->cache_control->hasNoCacheWithoutParameters() && !REFRESH_OVERRIDE(ignore_must_revalidate)) {
             debugs(22, 3, "Authenticated but server reply Cache-Control:no-cache (equivalent to must-revalidate)");
```

Comment:
I agree with Amos. Please note that, unlike old 064679ea374d2f58ae4660d9a2af213b9be24bba code, in modern code, `Cache-Control:no-cache` sets ENTRY_REVALIDATE_ALWAYS (rather than ENTRY_REVALIDATE_STALE that this PR is working on).

Before ignore-must-revalidate was removed in 064679ea374d2f58ae4660d9a2af213b9be24bba, the two flags were the same -- ENTRY_REVALIDATE_ALWAYS did not exist back then -- so this `if` statement had no choice but treat ignore-must-revalidate specially. We split ENTRY_REVALIDATE into two flags in fa83b76. Now, this `if` statement does not need to worry about ignore-must-revalidate.

```C++
if (ccNoCacheNoParams || ccPrivate)
    EBIT_SET(entry->flags, ENTRY_REVALIDATE_ALWAYS);
else if (ccMustRevalidate || ccSMaxAge)
    EBIT_SET(entry->flags, ENTRY_REVALIDATE_STALE);
```